dlk/python/dlk/templates/src/func/impl/quantized_conv2d_dim2col.cpp

```diff
@@ -101,11 +101,11 @@ void binary_convolution_cpu(QUANTIZED_PACKED input_channels[],
 
     for (T_UINT idx_k = 0; idx_k < bin_kernel_nwords; idx_k++) {
       for (T_UINT in_channel = 0; in_channel < num_in_channels; in_channel++) {
-        T_UINT in_data = input_channels[idx_in];
+        QUANTIZED_PACKED in_data = input_channels[idx_in];
 
         for (T_UINT k_pe = 0; k_pe < num_kernels; k_pe++) {
           T_UINT kernel_buf = kernel[k_pe * bin_kernel_nwords + idx_k];
-          T_INT xnor_result = pop_count(~(in_data ^ kernel_buf));
+          T_INT xnor_result = pop_count(~(static_cast<QUANTIZED_PACKED::T>(in_data) ^ kernel_buf));
```

Comment:
Could we make `kernel_buf` to be of type `QUANTIZED_PACKED`
As bitsize of `kernel_buf` should always match the size of `QUANTIZED_PACKED` storing it in `QUANTIZED_PACKED` seems to make sense to me...
It also allows us to rewrite expression above as
```cpp
T_INT xnor_result = (~(in_data ^ kernel_buf)).PopCount();
```
\**assuming we added `operator~` and `PopCount()`*

Reply:
Yes, `kernel_buf` should be `QUANTIZED_PACKED`, but this change require much implementation (python code should have information about packed or not).
I will implement in other PR.

Reply:
For now, you probably could just something like below, in this PR, and update python and other things in the other one.
```cpp
auto kernel_buf = QUANTIZED_PACKED(kernel[k_pe * bin_kernel_nwords + idx_k]);
```
:thinking:

Reply:
That's right.
Fixed.